File: node-ts/src/estimate.test.ts
import { calculateEstimate } from "./estimate.js";

if (calculateEstimate({ area: 10, unitPrice: 1000 }) !== 10000) {
  throw new Error("Test failed: valid case");
}

let threw = false;
try {
  calculateEstimate({ area: 0, unitPrice: 1000 });
} catch {
  threw = true;
}
if (!threw) {
  throw new Error("Test failed: invalid case");
}

console.log("OK");

File: go/cmd/estimate/estimate_test.go
package estimate

import "testing"

func TestCalculateOK(t *testing.T) {
	got, err := Calculate(Params{Area: 10, UnitPrice: 1000})
	if err != nil {
		t.Fatalf("expected nil error, got %v", err)
	}
	if got != 10000 {
		t.Fatalf("expected 10000, got %v", got)
	}
}

func TestCalculateNG(t *testing.T) {
	_, err := Calculate(Params{Area: 0, UnitPrice: 1000})
	if err == nil {
		t.Fatalf("expected error")
	}
}
